drivers/adc/adc_context.h

```diff
@@ -216,6 +216,10 @@ static inline void adc_context_start_read(struct adc_context *ctx,
 			adc_context_enable_timer(ctx);
 			return;
 		}
+	} else {
+		/* Explicitly overwrite in case previous read contained options */
+		ctx->options = (struct adc_sequence_options){ 0 };
+		ctx->sequence.options = NULL;
```

Comment:
It's not needed. If `ctx->sequence.options` is `NULL` (and it is set so by: https://github.com/zephyrproject-rtos/zephyr/blob/f5091058f337b39ab1207a76c517193a72fd29e3/drivers/adc/adc_context.h#L206 when `sequence->options` is `NULL`), a given driver should not refer to `ctx->options` at all (and if it does, it is a bug in that driver), so it does not really matter what is in that structure then.